### ISSUE TITLE
Reading configuration from secretsmanager.properties fails when running Spring Boot jar

### ISSUE THREAD

**Issue by ecimon** [aws/aws-secretsmanager-jdbc#11]
Related to https://github.com/aws/aws-secretsmanager-jdbc/issues/5, I'm trying to set the property `drivers.mysql.realDriverClass=com.mysql.cj.jdbc.Driver`. Putting this to `src/main/resources/secretsmanager.properties` works fine when running the app in IDE, but when I package it to Spring Boot jar with `spring-boot-maven-plugin` defaults, then `aws-secretsmanager-jdbc` cannot read it anymore and uses the default MySQL driver.

I noticed that the root cause is here https://github.com/aws/aws-secretsmanager-jdbc/blob/master/src/main/java/com/amazonaws/secretsmanager/util/Config.java#L70: `ClassLoader.getSystemResourceAsStream(resourceName)` returns `null` when running the packaged jar. Using `Thread.currentThread().getContextClassLoader().getResource("secretsmanager.properties")` seems to work fine in both IDE and jar, but I don't know if there are any side effects of this implementation.

As a workaround, I used `System.setProperty("drivers.mysql.realDriverClass", "com.mysql.cj.jdbc.Driver");` to set the property in code.

**Comment by willtong1234** [comment]
Have you tried updating the config loader to use a Spring utility in order to resolve resource locations (see https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/util/ResourceUtils.html)?

We likely will not be making this change to the github repo as loading files from they system classpath is a more generic solution that would work in most cases. I'll discuss with my team to ensure they all agree before closing out this issue.

**Comment by ecimon** [comment]
I tried loading the config file like Spring `ResourceUtils.getURL` would load it and it works. Internally, `getURL` (https://github.com/spring-projects/spring-framework/blob/master/spring-core/src/main/java/org/springframework/util/ResourceUtils.java#L129) calls `ClassUtils.getDefaultClassLoader()`, which in turn (https://github.com/spring-projects/spring-framework/blob/master/spring-core/src/main/java/org/springframework/util/ClassUtils.java#L180) calls first `Thread.currentThread().getContextClassLoader();`, then `ClassUtils.class.getClassLoader();` and finally `ClassLoader.getSystemClassLoader();` if the preceding calls do not work. That's how I discovered the `Thread.currentThread().getContextClassLoader().getResource("secretsmanager.properties")` method.

One option to gently update the `aws-secretsmanager-jdbc` config file loading would be to first try to load the file with `ClassLoader.getSystemResourceAsStream(resourceName)` and if that returns null then use `Thread.currentThread().getContextClassLoader().getResource(resourceName)`. This would preserve backwards-compatibility when the configuration is present in system classpath.
